### PR TITLE
feat: finish auth page with protected routes

### PR DESCRIPTION
**Pull request by Istiniel**
finish auth page with protected routes
fix git oauth redirect & fix min input length code in forms

### CODE REVIEW

File: public/locales/en/auth.json [hunk 1]
```diff
@@ -12,9 +12,10 @@
   "nopass": "Please enter password",
   "nologin": "Please enter login",
   "noemail": "Please enter Email",
-  "min": "At least 5 characters",
+  "min": "At least 8 characters",
```
Comment (AnAtoliy-AA) [inline public/locales/en/auth.json]: hardcoded number value

File: src/components/AuthForm/SignInForm/index.tsx [hunk 2]
```diff
@@ -1,47 +1,63 @@
...
+          required: 'noemail',
           pattern: {
-            value: /^(?=.*[A-Za-z0-9]$)[A-Za-z][A-Za-z\d.-]{0,19}$/,
-            message: 'wrongInput',
-          },
-          minLength: {
-            value: 5,
-            message: 'min',
+            value:
+              /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|.(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
```
Comment (AnAtoliy-AA) [inline src/components/AuthForm/SignInForm/index.tsx]: try to use this one `/^[^\s@]+@[^\s@]+.[^\s@]+$/`

File: src/hooks/useSendEmailVerification.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+export type SendEmailVerificationHook = [
+  () => Promise<boolean>,
+  boolean,
+  AuthError | Error | undefined,
+]
+
+const useSendEmailVerification = (auth: Auth): SendEmailVerificationHook => {
+  const [error, setError] = useState<AuthError>()
+  const [loading, setLoading] = useState<boolean>(false)
```
Comment (AnAtoliy-AA) [inline src/hooks/useSendEmailVerification.ts]: different namings in diff places
some strings before: `const [isLoading, setIsLoading] = useState(true)` but without type of state)

File: public/locales/en/auth.json (hunk 1 above)
Comment (Istiniel) [reply public/locales/en/auth.json]: fixed in the last commit

File: src/components/AuthForm/SignInForm/index.tsx (hunk 2 above)
Comment (Istiniel) [reply src/components/AuthForm/SignInForm/index.tsx]: fixed in the last commit

File: src/hooks/useSendEmailVerification.ts (hunk 3 above)
Comment (Istiniel) [reply src/hooks/useSendEmailVerification.ts]: partially fixed in the last commit

**Review by AnAtoliy-AA — APPROVED** [review]

**Pull request merged**